src/backend/distributed/replication/multi_logical_replication.c

```diff
@@ -291,33 +276,13 @@ LogicallyReplicateShards(List *shardList, char *sourceNodeName, int sourceNodePo
 	}
 	PG_CATCH();
 	{
+		/* We don't need to UnclaimConnections since we're already erroring out */
+
 		/*
-		 * Try our best not to leave any left-over subscription or publication.
-		 *
-		 * Although it is not very advisable to use code-paths that could throw
-		 * new errors, we prefer to do it here since we expect the cost of leaving
-		 * left-overs not be very low.
+		 * Drop temporary objects that were marked as CLEANUP_ON_FAILURE
+		 * or CLEANUP_ALWAYS.
 		 */
-
-		/* reconnect if the connection failed or is waiting for a command */
-		RecreateGroupedLogicalRepTargetsConnections(groupedLogicalRepTargetsHash,
```

Comment:
FinalizeOperationNeedingCleanupOnFailure only deals with placements, here we also deal with subs/pubs. Is that OK to remove here?

Reply:
`FinalizeOperationNeedingCleanupOnFailure` actually processes subs/pubs as well. Why do you think it doesn't deal with them? Is there a case that doesn't work for these objects?

Reply:
ah, I checked the main branch, this PR changes it, sorry for the confusion :)